### ISSUE TITLE
Delegates.

### ISSUE THREAD

**Issue by proofit404** [proofit404/stories#39]
For example, we are running story in the Celery task.

But this task was delayed from somewhere.

It may be a different story with its own execution path and context.

Showing this path and context in the called task can be really helpful to debug problems.

Should be supported by #38 

This is leading into the concept of `Delegate`s. `Canvas` is an example of the `Delegate`.

Generally speaking, we are trying to solve this problem:

```python
result = story1()
story2(result) # fails without knowing anything about story1 context.
```

If this code was written in this below, it will be way more easy to debug.

```python
d = Delegate(story2)
story1(d)
d() # Takes arguments from story1 context.
```

GraphQL mutations are perfect candidates for delegates.

They are executed sequentially one after another.

If mutation number 10 fails, we would like to see the execution path of the previous 9.

#### Parallel execution

As always, composition is a superior pattern & I'm glad that this feature was designed around it.

Delegates could be used to execute nested stories in parallel or concurrently.

Specific delegates implementation like `stories-asyncio` or `stories-celery` could run delegates any way they like.

The best part of that decision, all implementation specific details will be moved to the composition (Injector) step.

For example, `stories-asyncio` could execute nested stories using gather function. Outer story will be continue it's execution after it was suspended to execute delegate.

In comparison, `stories-celery` could execute nested stories using group part of chore canvas. Outer story will be head part of the same chore and will be scheduled as a separate task and will restore its state the same way nested stories did.

#### Distributed execution

State object store not only normalized values, but also it store initial data for each value. Usually, it is entity and a dictionary of its arguments.

In that case, `stories-celery` would serialize initial data as task argument.

Before delegate continue it's execution on another machine, contract will apply normalization and restore entity classes the same way it did in the first story.

That way we would use instances of our classes traveling over network without pickle.

**Comment by proofit404** [comment]
This is leading into the concept of `Delegate`s. `Canvas` is an example of the `Delegate`.

Generally speaking, we are trying to solve this problem:

```python
result = story1()
story2(result) # fails without knowing anything about story1 context.
```

If this code was written in this below, it will be way more easy to debug.

```python
d = Delegate(story2)
story1(d)
d() # Takes arguments from story1 context.
```

**Comment by proofit404** [comment]
GraphQL mutations are perfect candidates for delegates.

They are executed sequentially one after another.

If mutation number 10 fails, we would like to see the execution path of the previous 9.

**Comment by proofit404** [comment]
Superseded by #627
